### ISSUE TITLE
DXC doesn't support aggressive packing rule as a replacing

### ISSUE THREAD

**Issue by sixzone11** [microsoft/DirectXShaderCompiler#2888]
I've been noticed this when I profiled a difference in perf. between results from FXC and DXC.

As instance,
```
cbuffer Sample
{
    int4 __packed[16];
};

static int arrayReallyWant[64] = (int[64])__packed;
```

A result from DXC shows fool performance. There are more VRAM accesses and lower cache hits than from FXC.
I saw the DXIL and found that those lines are translated as simply assinging each slot to a local variable. And as I known, define too many local variable cause those not placed in register fall to global memory space. (on NVIDIA GPU. I don't know on any other cases.)

In specially, it cause chaos because aggressive packing is described in MSDN without any mension about performance.

So I think there should be an option to avoid using unrecognised local variable. Those statements may be replaced by accessing cbuffer directly.

**Comment by pow2clk** [comment]
repro shader: 

cbuffer Sample
{
    int4 __packed[16];
};

static int arrayReallyWant[64] = (int[64])__packed;

int4 main() : SV_Target {
    return arrayReallyWant[0];
}

There's a lot of unnecessary copying going on here.

**Comment by pow2clk** [comment]
@python3kgae fixed this with #3483
